### PR TITLE
Feature/header part d record details fragment

### PR DESCRIPTION
**Pull request by schuetzcarl**
Done in that PR : 
- Align the Roadbook theme with the `DisplayFragment`'s one & add Dialog animation.
- New `DRecordInfoFragment` class
- `DRecordDetailsFragment` is now implemented with the ViewPager2 and TabLayout
- Fix of some incoherence when drag & dropping some record of the RB : 
Resolved by replacing the `AsyncListDiffer` with a simple `MutableList` in `RoadBookViewAdapter`
The normal solution of the incoherence leads me to that well known bug :
https://stackoverflow.com/questions/65804749/drag-dropping-the-first-item-of-the-recyclerview-moves-several-random-position
That's why I finally decide to get rid of the `AsyncListDiffer` which is not important for the roadbook as a list of `DestinationRecord` is not heavy to update.

### CODE REVIEW

File: app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt [hunk 1]
```diff
@@ -1,44 +1,79 @@
...
         }
 
+        val toolbar = requireActivity().findViewById<Toolbar>(R.id.toolbar)
+        toolbar.setTitle("Dest. Record : $destID")
+
+        viewPager = view.findViewById(R.id.viewPager)
+        val adapter = DetailsFragmentsAdapter(this)
+        adapter.addFragment(DRecordInfoFragment(rec))
+        adapter.addFragment(RouteFragment())
+        adapter.addFragment(DirectoryFragment()) // To be replaced with ContactDetailsFragment()
```
Comment (yanvow) [inline app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt:42]: Should we add a TODO?

File: app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt [hunk 2]
```diff
@@ -1,44 +1,79 @@
...
+        super.onViewCreated(view, savedInstanceState)
+
+        val tabLayout: TabLayout = view.findViewById(R.id.tabLayout)
+        TabLayoutMediator(tabLayout, viewPager) { tab, position ->
+            val tabFrag = TabFragment.values()[position]
+            tab.setIcon(tabFrag.iconID)
+        }.attach()
+    }
+    enum class TabFragment(val iconID: Int) {
+        INFO(R.drawable.baseline_info_24),
```
Comment (yanvow) [inline app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt:59]: We should probably standardize the name of our icons.

**Review by yanvow — APPROVED** [review]
Overall very nice code and good tests.
Moreover, you also changed some old code to make it more understandable and modulable.
However, some minor standadization of names and refactoring would be nice.

File: app/src/main/res/drawable/baseline_map_24.xml [hunk 3]
```diff
@@ -0,0 +1,5 @@
+<vector android:height="24dp" android:tint="#000000"
```
[the remark this answers is not in the document]
Comment (schuetzcarl) [reply app/src/main/res/drawable/baseline_map_24.xml:1]: I needed it in the Drawable, but didn't found an existing one. That's why I created a new vector with that icon.
Then, I didn't found the same one when seraching with a "map" entry. Maybe I just miss it.

File: app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt (hunk 1 above)
Comment (schuetzcarl) [reply app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt:42]: I think it doesn't need. It's clearly obvious as there is a designated task in the current Sprint for that. And it's not a hidden code place. The task programmer will be forced to arrive there and see that comment.

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt [hunk 4]
```diff
@@ -482,6 +481,7 @@ class RoadBookFragmentTest {
         // Navigate outside and come back
         onView(withId(R.id.drawer_layout))
             .perform(DrawerActions.open())
+        onView(withId(R.id.routeFragment))
```
[the remark this answers is not in the document]
Comment (schuetzcarl) [reply app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/RoadBookFragmentTest.kt:484]: It isn't :')

File: app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragmentTest.kt [hunk 5]
```diff
@@ -87,4 +92,44 @@ class DRecordDetailsFragmentTest {
...
+        toFragment()
+        onView(withId(R.id.fragment_directory_directors_parent)).check(doesNotExist())
+        onView(withId(R.id.viewPager)).perform(swipeLeft())
+        onView(withId(R.id.viewPager)).perform(swipeLeft())
+        onView(withId(R.id.fragment_directory_directors_parent)).check(matches(isDisplayed()))
+    }
+
+    // I think block in the CI due to the camera authorizations however it begins to be @Jules part,
+    // and maybe a different Fragment will be here.
+    private fun swipeLeftThreeTimesDisplaysPicture() {
```
[the remark this answers is not in the document]
Comment (schuetzcarl) [reply app/src/androidTest/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragmentTest.kt:115]: Yes that's why I commented that.
As it is already written, I think we can keep it. In the source code I would avoid that. But in the test code, I think it is ok.

File: app/src/main/res/values-night/themes.xml [hunk 6]
```diff
@@ -14,4 +14,9 @@
         <!-- Customize your theme here. -->
         <item name="cardViewStrokeColor">@color/cardview_stroke_color_dark_mode</item>
     </style>
+
+    <style name="MyDialogAnimation">
```
[the remark this answers is not in the document]
Comment (schuetzcarl) [reply app/src/main/res/values-night/themes.xml]: Sure. It's done

File: app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt (hunk 2 above)
Comment (schuetzcarl) [reply app/src/main/java/com/github/factotum_sdp/factotum/ui/roadbook/DRecordDetailsFragment.kt]: I renamed the baseline_xxxx_24 ones, but for the older ones, will see with the others in a next meeting, on which standards you want to seat

**Pull request merged**